<!DOCTYPE html>
<html>
  <head>
    <title>Testing pleiotropy in multiparental populations</title>
    <meta charset="utf-8">
    <meta name="author" content="Frederick Boehm" />
    <meta name="date" content="2019-03-06" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="slides.css" type="text/css" />
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Testing pleiotropy in multiparental populations
### Frederick Boehm
### March 6, 2019

---













# Introduction

- 10,000+ traits with RNA sequencing and mass spectrometry 

- Multiparental populations offer high-resolution QTL mapping

- New analysis tools, such as a pleiotropy test for multiparental populations, are needed 


???

- Together, high-dimensional traits and multiparental populations can inform complex trait genetics

---

# Pleiotropy &amp; *Agouti* gene

&lt;img src="https://upload.wikimedia.org/wikipedia/commons/4/4d/Agouti_Mice.jpg" width="500" /&gt;

.small[https://upload.wikimedia.org/wikipedia/commons/4/4d/Agouti_Mice.jpg]

???

- Agouti protein antagonizes `\(\alpha\)`-melanocyte-stimulating hormone  
    - prevents melanocyte production of melanin  
    - disrupts melanocortin receptor signaling in the brain  

- Agouti viable yellow ($A^{vy}$) mice    

---

# Jiang and Zeng  (1995)  test

- Two-parent crosses    

- Applies to two traits that co-map     

- `\(H_0\)`: Pleiotropy    

- `\(H_A\)`: Two separate QTL    



---

# Jiang and Zeng  (1995)  test


- Perform a two-dimensional QTL scan     

- `\(vec(Y) = Xvec(B) + vec(E)\)`     

- Calculate likelihood ratio test statistic     

???

- Y: 2 traits  
- X: genotype probs for 2 markers  
- B: founder allele effects  
- E: random errors  
- vec: stack columns

- 2d scan involves all ordered pairs of markers


---

# Jiang and Zeng  (1995)  test



![](testing-pleiotropy_files/figure-html/encircle-1.png)&lt;!-- --&gt;

???

- Compare with a t-test where you test a point hypothesis as a null and the alternative is the real number line


---

# Diversity Outbred Mice


  

&lt;img src="figs/ccmice.png" width="400px" /&gt;

???

- Multiparental population arising from 8 inbred lines    

- Each DO mouse is genetically unique  
- 3 wild-derived lines
- 3 subspecies: *M. m. domesticus*, *M. m. musculus*, and *M. m. castaneus*



---

# Multiparental populations

&lt;img src="testing-pleiotropy_files/figure-html/mppfig-1.png" width="700px" /&gt;


???


- start with 8 inbred founder lines  
- inbred lines are color-coded  
- each subject represented by one homologous chromosome pair  
- over successive generations of outbreeding, subjects have smaller and smaller contiguous regions from a single inbred line  



---

# Challenges in multiparental populations

- Complex patterns of relatedness    

- Multiple founder lines

- Determining statistical significance    


---

# Challenges in multiparental populations

- Complex patterns of relatedness    

.soln[Multivariate random effects]

- Multiple founder lines

.soln[Fixed effect for each founder allele]    

- Determining statistical significance    

.soln[Parametric bootstrap]    



---

# Test procedure

- Model: `$$vec(Y) = Xvec(B) + vec(G) + vec(E)$$` 
    
- Calculate likelihoods

???

- `\(G\)`: Polygenic random effects   
    
- `\(E\)`: Random errors
    


---

# Test procedure

- Test statistic: `$$- \log \frac{\max (\text{likelihood under pleiotropy})}{\max (\text{likelihood for separate QTL})}$$`    

- Parametric bootstrap to get p-value    


---

# Application

- Logan, et al.  (2013)  and  Recla, et al.  (2014)  studied 261 Diversity Outbred mice    

- Two traits map to Chr 8:

    - "hot plate latency" (57 cM)    

    - "percent time in light" (55 cM)

???

Do they share a single QTL? 

---

# QTL scan results

&lt;img src="figs/genomewide_lods_10-22.svg" height="500px" /&gt;

---

# QTL scan results

&lt;img src="figs/chr8-lods.svg" height="500px" /&gt;




---

# Allele effects plots  


&lt;img src="figs/coefs.svg" width="500px" /&gt;

???

- Similar allele effects are consistent with pleiotropy

---

# Profile LOD

`$$\text{profile LOD}_{\text{trait 1}}(\lambda_1) = max_{\lambda_2}LOD(\lambda_1, \lambda_2)$$`


&lt;img src="testing-pleiotropy_files/figure-html/profile-trace-1.png" width="400" /&gt;

---

# Profile LOD plot

![](figs/profile.svg)&lt;!-- --&gt;




---

# Test results

- p = 0.11 (1000 bootstrap samples)


---

# Conclusions

- Weak evidence for two separate QTL

    - One QTL affects "distance traveled in light"    
    
    - Second QTL affects "hot plate latency"


---

class: center, middle, inverse, title-slide

# Mediation analysis and pleiotropy testing


---

# Mediation analysis

- Is one molecule an intermediate in a causal pathway?  

---

# Mediation analysis example 

- QTL near Dhtkd1 gene affects Dhtkd1 RNA levels and DHTKD1 protein levels

- Is Dhtkd1 RNA an intermediate between QTL and DHTKD1 protein levels? 


![](figs/central_dogma-Dhtkd1.svg)&lt;!-- --&gt;

---

# Mediation analysis methods

- LOD score with and without conditioning on candidate intermediate

- LOD difference   

- LOD difference proportion    


---

# Dissecting an eQTL hotspot

- Keller, et al.  (2018)  identified 147 nonlocal traits that map to Chr 2 (163.5 Mb to 167.5 Mb)  
 
- *Hnf4a* expression levels mediate 88 of 147 traits  

- Do other local genes act as intermediates? 


???

- 378 DO mice  

---

# Dissecting an eQTL hotspot

- 13 local gene expression traits as candidate mediators

- 147 nonlocal gene expression traits

- Pair one local trait with one nonlocal trait




---

# Dissecting an eQTL hotspot

&lt;img src="figs/lod-diff-prop-v-lrt.svg" width="500" /&gt;


---

# Dissecting an eQTL hotspot

&lt;img src="figs/Hnf4a-lod-diff-prop-v-lrt.svg" width="500" /&gt;




---

# Dissecting an eQTL hotspot


![](figs/12local-facet_grid-no-strip-3-by-4.svg)&lt;!-- --&gt;

---

# Dissecting an eQTL hotspot

&lt;img src="figs/nonlocal-4-panels-no-strip-wide.svg" width="650" /&gt;

---

# Conclusions

- Pleiotropy test &amp; mediation analysis ask distinct questions

- Both are useful when dissecting eQTL hotspots

???

- Pleiotropy test informs about the number of QTL in a hotspot, even when there is no clear mediator   







---

class: center, middle, inverse, title-slide

# qtl2pleio R package development


---

# Design elements

- Written mostly in R    
  
  - Some calculations in C++, via Rcpp package  
  
  - C++ Eigen library and RcppEigen for matrix algebra  
  
  - User interacts only with R functions  

---

# Design elements

- Tutorial vignettes demonstrate functionality  

- Unit tests included via `testthat` R package  

- Documentation with `roxygen2` R package

- Package website with `pkgdown` R package

---

# R packages book

- Hadley Wickham's "R packages" book as a guide  

&lt;img src="figs/rpackages-cover.png" width="300px" /&gt;



---

# Open-source R package

- Hosted on github  

- Install via `devtools` R package  

- `git` commits publicly visible  

- Travis CI for testing after every commit  


---

# Binder integration

- Enhances reproducibility  

- Uses Docker images 

- Downloads R packages for a specified date (or `git` commit on Github)  


---

# Binder integration: 2 new files

- install.R

    - contains code to install R packages

- runtime.txt

    - contains one line of text to indicate date for CRAN packages
    


---

# Contact information

- frederick.boehm@gmail.com   

- https://fboehm.us/     

- qtl2pleio R package: [https://github.com/fboehm/qtl2pleio](https://github.com/fboehm/qtl2pleio)

- Biorxiv pre-print: [http://bit.ly/boehm-biorxiv](http://bit.ly/boehm-biorxiv)



---

# References

Jiang, C. and Z. Zeng (1995). "Multiple trait analysis of genetic
mapping for quantitative trait loci." In: _Genetics_ 140.3, pp.
1111-1127.

Keller, M. P, D. M. Gatti, et al. (2018). "Genetic Drivers of
Pancreatic Islet Function". In: _Genetics_, pp. genetics-300864.

Logan, R. W, R. F. Robledo, et al. (2013). "High-precision genetic
mapping of behavioral traits in the diversity outbred mouse
population". In: _Genes, Brain and Behavior_ 12.4, pp. 424-437.

---

# References

Recla, J. M, R. F. Robledo, et al. (2014). "Precise genetic
mapping and integrative bioinformatics in Diversity Outbred mice
reveals Hydin as a novel pain gene". In: _Mammalian genome_
25.5-6, pp. 211-222.
    </textarea>
<script src="libs/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"slideNumberFormat": "%current%",
"ratio": "4:3"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_HTMLorMML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
